src/windows/gfx/d3d11-ctx.c

```diff
@@ -45,8 +58,185 @@ static void d3d11_ctx_get_size(struct d3d11_ctx *ctx, uint32_t *width, uint32_t
 	*height = rect.bottom - rect.top;
 }
 
+static void mty_validate_format_colorspace(struct d3d11_ctx *ctx, MTY_ColorFormat format, MTY_ColorSpace colorspace, DXGI_FORMAT *format_out, DXGI_COLOR_SPACE_TYPE *colorspace_out)
+{
+	DXGI_FORMAT format_new = DXGI_FORMAT_R8G8B8A8_UNORM;
+	DXGI_COLOR_SPACE_TYPE colorspace_new = DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709;
+
+	// Use the last known value if unspecified
+	if (format == MTY_COLOR_FORMAT_UNKNOWN) {
+		format_new = ctx->format;
+	}
+	if (colorspace == MTY_COLOR_SPACE_UNKNOWN) {
+		colorspace_new = ctx->colorspace;
+	}
+
+	switch (format) {
+		case MTY_COLOR_FORMAT_RGBA16F:
+			format_new = DXGI_FORMAT_R16G16B16A16_FLOAT;
+			break;
+		case MTY_COLOR_FORMAT_RGB10A2:
+			format_new = DXGI_FORMAT_R10G10B10A2_UNORM;
+			break;
+	}
+
+	switch (colorspace) {
+		case MTY_COLOR_SPACE_SRGB:
+			colorspace_new = DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709;
+			break;
+		case MTY_COLOR_SPACE_SCRGB_LINEAR:
+			colorspace_new = DXGI_COLOR_SPACE_RGB_FULL_G10_NONE_P709;
+			break;
+		case MTY_COLOR_SPACE_HDR10:
+			colorspace_new = DXGI_COLOR_SPACE_RGB_FULL_G2084_NONE_P2020;
+			break;
+	}
+
+	// Ensure that the format and colorspace are a valid pairing
+	switch (colorspace_new) {
+		case DXGI_COLOR_SPACE_RGB_FULL_G10_NONE_P709: {
+			if (format_new != DXGI_FORMAT_R16G16B16A16_FLOAT) {
+				MTY_Log("Format 0x%X is not meant for colorspace 0x%X. Forcing format to 0x%X.", format_new, DXGI_COLOR_SPACE_RGB_FULL_G10_NONE_P709, DXGI_FORMAT_R16G16B16A16_FLOAT);
+				format_new = DXGI_FORMAT_R16G16B16A16_FLOAT;
+			}
+			break;
+		}
+		case DXGI_COLOR_SPACE_RGB_FULL_G2084_NONE_P2020: {
+			if (format_new != DXGI_FORMAT_R10G10B10A2_UNORM) {
+				MTY_Log("Format 0x%X is not meant for colorspace 0x%X. Forcing format to 0x%X.", format_new, DXGI_COLOR_SPACE_RGB_FULL_G2084_NONE_P2020, DXGI_FORMAT_R10G10B10A2_UNORM);
+				format_new = DXGI_FORMAT_R10G10B10A2_UNORM;
+			}
+			break;
+		}
+		default:
+			break;
+	}
```

Comment:
Wouldn't it be better if the colorspace would be set differently, since the colors would be "more" correct if the colorspace would be wrong, instead of the format?
Also remove the default case here.